### ISSUE TITLE
Synchronise (2 way perhaps) Planner with GitHub issues

### ISSUE THREAD

**Issue by pradeepto** [fabric8io/fabric8-planner#1879]
Deploy projects are upstream on Github. [Kompose](https://github.com/kubernetes-incubator/kompose/issues) and [OpenCompose](https://github.com/redhat-developer/opencompose/issues).

For us (Deploy) to use planner productively, a 2-way sync with GitHub a great feature to have. Let me know how we can collaborate to spec this feature out in case that is not done yet. There are others like the Che folks who need this feature as well.

**Comment by pradeepto** [comment]
Here are my thoughts: 

# Scenario 1:
Bulk import - My project is on GH and I considering using OSIO Planner. I would like to import all my existing issues as "work items" and (maybe) the associated conversations to Planner.

# Scenario 2:
Individual import - My project is already on Planner and GH. There are upstream users who still use GH to file issues or comment on issues (maybe because they don't have access to planner or don't want to use planner or they use some kind of API to update GH.) I would like the following to happen:

a) every time a new issue is filed on GH project that is synced with Planner, the new issue should be automatically synced with Planner as a "work item".
b) every time there is a new comment or action on GH on an already synced issue, the comment or action to should be synced with Planner.
c) everytime I comment on a planner work item, it should be propagated to GH issue.

There is a definite need for a configuration for how "deep" these import/sync should be? These are perhaps an UX issue.

**Comment by maxandersen** [comment]
we have a notion of remote work items to be able to track things like external github issues.

in that notion was the idea of being able to at least track status, title and description (possibly assignee) as synced from github - but *NOT* try and sync things like comments since that is going to just create double noise with respect to notifications of updates and whatnot.

**Comment by pradeepto** [comment]
@maxandersen That makes sense and should work, IMO.

**Comment by debloper** [comment]
# Scenario 3
Issues with certain `labels` are *all* imported by planner. A good middle ground between the first 2 scenario.

**Comment by pradeepto** [comment]
Just importing is doesn't work. We have to manage bits at two places. Synchronisation is important for us.

**Comment by gorkem** [comment]
@maxandersen Why should  planner create notifications for remote WIs? I think those are features that should be left to the remote tracker.

**Comment by maxandersen** [comment]
> @maxandersen Why should planner create notifications for remote WIs? I think those are features that should be left to the remote tracker.

yes,  which is why I would not *import* the comments from github into the remote work item.

remember the remote work item would be something you put into a sprint in planner - you would want a notification on that. But you doing a comment on issue in github on the actual github issue is not relevant for the remote work item in planner. In other words - very much like done for Eclipse Bugzilla in our bugzilla jira importer and board planning (hope that makes sense)

**Comment by maxandersen** [comment]
> Issues with certain labels are all imported by planner. A good middle ground between the first 2 scenario.

what is implemented is "issues found by a github query" (which can be used to do certain labels).

What is missing though is how github issues gets mapped into the work item type system. Which is longer and more hard problem to solve and where we paused it at for now.
